Lullabot/media_mpx
# Initial media entity for media objects

Depends on #4 .

This PR adds support for configuring an mpx media type for media - the core object in mpx we get video data from.

To configure:

1. Create an mpx user and account under `/admin/config/media/mpx`.
1. Create a new media type under `/admin/structure/media` and select 'mpx Media' as the type. *Set a label first* as there is a core bug causing a 503 on the `#ajax` callback if the machine name isn't set (I need to find / create that bug).

Most complex fields are not supported yet - just primitive types and dates.

<img width="810" alt="edit test drush site-install 2018-04-05 15-13-00" src="https://user-images.githubusercontent.com/255023/38386798-dec603c4-38e3-11e8-8936-f23c6cf97c0a.png">

To test field mapping:

1. Add a datetime field to the media type you created.
1. Back at the main configuration form, set `The date that this content becomes available for playback.` to the field you created.
1. Add a new media item. In the `mpx Media` field, paste a URL such as `https://data.media.theplatform.com/media/data/Media/12345` with one valid for your configured mpx account.
1. On save, the date/time field should be populated.

Open questions:
- [ ] Should I remove the Account configuration entity, and merge it directly into the media type configuration? I feel like it helps make the UI simpler, but now that I think of it I wonder if it's always going to be a one-to-one relationship between accounts and media types.
